### ISSUE TITLE
Generator prototype ignored

### ISSUE THREAD

**Issue by skyh** [microsoft/TypeScript#36086]
**TypeScript Version:**  3.8.0-dev.20200107

**Search Terms:** generator, iterator, prototype, inherit

**Code**

```ts
function* gen() {
    while (true) {
        yield "something";
    }
}

gen.prototype.test = () => {
    console.log("OK");
};

gen().test();
```

**Expected behavior:** No errors

**Actual behavior:** Error `TS2339: Property 'test' does not exist on type 'Generator<string, void, unknown>'`

**Playground Link:**  [Provided](https://www.typescriptlang.org/play/?target=99&ts=3.8.0-dev.20200107#code/GYVwdgxgLglg9mAVAAgOYFMwAoCUyDeAsAFDJnIDuAFjADbrJZQBOI6eRp53AnjOrQAmyAEQBnOAFt0UGmFQiA3CW4BfEuuIkMYAHQAHZnCjGe+9LqjoxUZAF5GeOwD4CK8hAQT6u2nFRYIgDyANIiOMrEqpHamLiW1lC4isgA9KnIAKLMRszIACoAygBMAMylAJwAXMgACkbmzFA8yADkVjatyIJw1shgxsjoAB4wNsgIyM3mbQDimOjMAIYmzAA8Nsww8gA0yABucDCCe+AA1gMUYM6tuipAA)

**Comment by leadpipe** [comment]
I asked about this very issue on StackOverflow, and got a helpful workaround:

https://stackoverflow.com/questions/67205298/is-it-possible-to-correctly-type-a-generator-function-with-extra-behavior-in-typ/67205596#67205596
